webapp/src/shared/context/SearchContext.tsx

```diff
@@ -4,13 +4,15 @@ import { useData } from '../hooks/UseData';
 
 type SearchContextProps<T> = {
   children: React.ReactNode;
-  fetchFn: (requestParams: {}) => Promise<T[]>;
+  fetchFn: <RequestType extends Query>(
+    requestParams: RequestType,
+  ) => Promise<T[]>;
```

Comment:
praise: Thanks for improving the typings and good idea to extend the `Query` type

Reply:
Thanks!
I've gona ahead and placed in the shared types file.